clients/src/main/java/org/apache/kafka/common/requests/AbstractResponse.java

```diff
@@ -51,8 +53,8 @@ final ByteBuffer serializeWithHeader(ResponseHeader header, short version) {
     }
 
     // Visible for testing
-    final ByteBuffer serialize(short version) {
-        return MessageUtil.toByteBufferAccessor(data(), version).buffer();
+    final ByteBufferAccessor serialize(short version) {
```

Comment:
should we return the interface (Readable/ReadableBuf) instead?

Reply:
It was like that in the Request PR, but as feedback I was asked to change that, so kept it same in this PR.